pkg/git/git.go

```diff
@@ -110,14 +110,46 @@ func (cloneOpts *CloneOptions) Validate() error {
 	return nil
 }
 
+// CheckoutFromBare checks out the specified reference from a bare repository
```

Comment:
I have left this function commented because I need review from @zong-zhe over the checkout implementation from bare, which I don't feel would be achieved something like that I did.